src/node/che-workspace-hosted-plugin-uri-post-processor.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (C) 2018 Red Hat, Inc. and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { injectable } from "inversify";
+import URI from "@theia/core/lib/common/uri";
+import { HostedPluginUriPostProcessor } from "@theia/plugin-ext";
+import WorkspaceClient, { IRemoteAPI, IWorkspace, IServer, IRestAPIConfig } from '@eclipse-che/workspace-client';
+
+@injectable()
+export class CheWorkspaceHostedPluginUriPostProcessor implements HostedPluginUriPostProcessor {
+    protected restApiClient: IRemoteAPI;
+
+    constructor() {
+        const restAPIConfig: IRestAPIConfig = {};
+        restAPIConfig.baseUrl = process.env.CHE_API;
+        const token = process.env.CHE_MACHINE_TOKEN;
+        if (token) {
+            restAPIConfig.headers = {};
+            restAPIConfig.headers['Authorization'] = token;
+        }
+        this.restApiClient = WorkspaceClient.getRestApi(restAPIConfig);
+    }
+
+    async processUri(uri: URI): Promise<URI> {
+        const hostedPluginTheiaInnstanceServer = await this.getHostedPluginTheiaInnstanceServer();
```

Comment:
somehow there are too many n in 'instance' getHostedPluginTheiaInnstanceServer

Reply:
Thanks